MM20605: Migrate 'components/popover_list_members' module and associated tests to TypeScript

sinansonmez opened the pull request:
#### Summary
This migrates 'components/popover_list_members' module and associated tests to TypeScript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13469

#### Screenshots
not applicable

#### Release Note
```release-note
NONE
```

mattermod commented:
Hello @sinansonmez,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

sinansonmez commented on components/popover_list_members/index.ts:
```diff
@@ -2,27 +2,50 @@
     return createSelector(
         'makeSortUsersByStatusAndDisplayName',
-        (state, channelId) => doGetProfilesInChannel(state, channelId, true),
+        (state: GlobalState, channelId: string) => doGetProfilesInChannel(state, channelId, {active: true}),
```
I assume true is referring to `active` in `Filters` type. Please let me know if my assumption is correct

michelengelen replied:
correct

sinansonmez commented on components/popover_list_members/index.ts:
```diff
@@ -33,9 +56,9 @@ function makeMapStateToProps() {
+    return function mapStateToProps(state: GlobalState, ownProps: Props) {
         const stats = getAllChannelStats(state)[ownProps.channel.id] || {};
-        const users = doGetProfilesInChannel(state, ownProps.channel.id, true);
+        const users = doGetProfilesInChannel(state, ownProps.channel.id, {active: true});
```
I assume true is referring to `active` in `Filters` type. Please let me know if my assumption is correct

michelengelen replied:
👍🏼

hmhealey replied:
Is this a change in behaviour from the previous version? It looks like passing `true` might have been the same as not filtering the result here, so changing it to `{active: true}` might start filtering out inactive users.

sinansonmez replied:
Actually, this had no effect. As discussed in other comment `users` prop wasn't being used in the component so I removed it from here as well.

michelengelen commented on components/popover_list_members/index.ts:
```diff
@@ -2,27 +2,50 @@
+
+type Actions = {
+    openModal: <P>(modalData: ModalData<P>) => void;
+    loadProfilesAndStatusesInChannel: (channelId: string, page: number, perPage: number | undefined, sort: string, options: {active: boolean}) => void;
```
out of scope for this PR, but I prefer to have optional parameters (in this case `perPage`) coming after the mandatory ones.

```suggestion
    loadProfilesAndStatusesInChannel: (channelId: string, page: number, perPage?: number, sort: string, options: {active: boolean}) => void;
```

sinansonmez replied:
I reverted this because when I make `perPage` optional as you said I need to put after required ones due to TypeScript rule. When I do this, I mix the order in [UserActions](https://github.com/mattermost/mattermost-webapp/blob/1ab8a75409c4ea71844fa92a8f9fdf2f748f52bd/actions/user_actions.jsx#L39). Any idea how to address this without changing  `UserActions`?

michelengelen replied:
in that case we could just leave it as is, as the person writing the `UserAction` was not paying attention. I will create a follow-up ticket for this to re-visit this in the future.

michelengelen replied:
well... i should have looked at the function definition before writing that comment ... it is written as it should be: all of the optional params are at the end.

just type it as follows:

```
loadProfilesAndStatusesInChannel: (channelId: string, page?: number, perPage?: number, sort?: string, options?: {active: boolean}) => void;
```

sinansonmez replied:
Ahh I see your point. As all params except `channelId` have default value, we can mark them optional with `?`.  Done

michelengelen commented on components/popover_list_members/index.ts:
```diff
@@ -2,27 +2,50 @@
 import {sortUsersByStatusAndDisplayName} from 'utils/utils.jsx';
 
-import PopoverListMembers from './popover_list_members.jsx';
+import PopoverListMembers, {Props} from './popover_list_members';
```
nicely sorted imports! 🚀

michelengelen commented on components/popover_list_members/index.ts:
```diff
@@ -2,27 +2,50 @@
     return createSelector(
         'makeSortUsersByStatusAndDisplayName',
-        (state, channelId) => doGetProfilesInChannel(state, channelId, true),
+        (state: GlobalState, channelId: string) => doGetProfilesInChannel(state, channelId, {active: true}),
```
not really needed, but to be on the safe side we could add `Channel['id']` here as type as well instead of string:

```suggestion
        (state: GlobalState, channelId: Channel['id']) => doGetProfilesInChannel(state, channelId, {active: true}),
```

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -14,34 +13,46 @@ import {localizeMessage} from 'utils/utils.jsx';
-        }).isRequired,
-        sortedUsers: PropTypes.array,
-        addMembersABTest: PropTypes.string,
+import {RelationOneToOne} from 'mattermost-redux/types/utilities';
```
also not related, but IMO the `RelationOneToOne` is not very helpful. all it basically does is checking for the first generic type to have the property `id` and assign a new generic property (which always will be `string`) to the second generic property as seen [here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wFgAoCmATzCUSQBsUZgIA7AeXaQBUJuSADwBROEgAeMJOwAmAZzgBvYLIBcceTCjB2AcwC+AGji8AfHAC8yinDtwA2hLi64Ih-lX4Aut428AbgoDIMpyGjo4AFV5JCgrZVUNLR19ALh2FBAkZO1dPQCDCgo0Di04aS0ARg1kZlYOQX5BIRi4k3YAVxAAIziLaxV1OCqAJhCSsvhKmFHaphY2Lh5mnlbYqA7uvqgBxOGxjqycggApCAALdnwQoA)

sinansonmez replied:
Thanks for additional info

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -14,34 +13,46 @@ import {localizeMessage} from 'utils/utils.jsx';
+    users: UserProfile[];
+    memberCount: number;
+    currentUserId: string;
+    teamUrl?: string;
```
could we sort optional properties to the end as well please?

sinansonmez replied:
Done

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -80,7 +91,7 @@ export default class PopoverListMembers extends React.PureComponent {
     };
 
-    showMembersModal = (e) => {
+    showMembersModal = (e: { preventDefault: () => void }) => {
```
this can be typed with one of Reacts built in Event-types

sinansonmez replied:
Done

michelengelen commented on components/popover_list_members/popover_list_members_item/popover_list_members_item.tsx:
```diff
@@ -0,0 +1,106 @@
+        props.onItemClick(props.user);
+    };
+
+    if (!props.user) {
+        return null;
+    }
+
+    if (!props.displayName) {
+        return null;
+    }
```
this can be combined:

```suggestion
    if (!(props.user || props.displayName)) {
        return null;
    }
```

michelengelen commented on components/popover_list_members/popover_list_members_item/popover_list_members_item.tsx:
```diff
@@ -0,0 +1,106 @@
+        return null;
+    }
+
+    let messageIcon;
+    if (props.showMessageIcon) {
+        messageIcon = (
+            <MessageIcon aria-hidden='true'/>
+        );
+    }
```
```suggestion
    const messageIcon = props.showMessageIcon ? (
        <MessageIcon aria-hidden='true'/>
    ) : null;
```

michelengelen commented on components/popover_list_members/popover_list_members_item/popover_list_members_item.tsx:
```diff
@@ -0,0 +1,106 @@
+        );
+    }
+
+    let sharedIcon;
+    if (props.user.remote_id) {
+        sharedIcon = (
+            <SharedUserIndicator
+                className='shared-user-icon'
+                withTooltip={true}
+            />
+        );
+    }
```
```suggestion
    const sharedIcon = props.user.remote_id ? (
        <SharedUserIndicator
            className='shared-user-icon'
            withTooltip={true}
        />
    ) : null;
```

michelengelen commented on components/popover_list_members/popover_list_members_item/popover_list_members_item.tsx:
```diff
@@ -0,0 +1,106 @@
+            />
+        );
+    }
+    const botClass = props.user.is_bot ? ' more-modal__row--bot' : '';
```
see comment below

```suggestion

```

michelengelen commented on components/popover_list_members/popover_list_members_item/popover_list_members_item.tsx:
```diff
@@ -0,0 +1,106 @@
+            onClick={handleClick}
+        >
+            <ProfilePicture
+                src={Client4.getProfilePictureUrl(props.user.id, props.user.last_picture_update)}
```
we could probably fetch this earlier and maybe store it in a memo or ref to prevent extensive (re-)fetching

sinansonmez replied:
Sorry, I couldnt understand your suggestion. Are you suggesting to use `useMemo` hook like:

`const memoizedValue = useMemo(() => Client4.getProfilePictureUrl(props.user.id, props.user.last_picture_update), [props.user]);`

sinansonmez replied:
Done

michelengelen commented on components/popover_list_members/popover_list_members_item/popover_list_members_item.tsx:
```diff
@@ -0,0 +1,106 @@
+};
+
+export default function PopoverListMembersItem(props: Props) {
+    const handleClick = () => {
+        props.onItemClick(props.user);
+    };
```
```suggestion
    const handleClick = () => props.onItemClick(props.user);
```

michelengelen commented on components/popover_list_members/popover_list_members_item/popover_list_members_item.tsx:
```diff
@@ -0,0 +1,106 @@
+            data-testid='popoverListMembersItem'
+            tabIndex={0}
+            aria-label={props.displayName.toLowerCase()}
+            className={'more-modal__row' + botClass}
```
to align this with other cases could we please use `classNames` package here?

```suggestion
            className={classNames('more-modal__row', {botClass: props.user.is_bot})}
```

michelengelen commented on components/popover_list_members/popover_list_members_item/popover_list_members_item.tsx:
```diff
@@ -0,0 +1,106 @@
+                    userID={props.user.id}
+                    showTooltip={true}
+                    emojiSize={15}
+                    emojiStyle={{
+                        marginBottom: -3,
+                        marginLeft: 4,
+                    }}
```
could we please add this as a `const` above, so it does not create a new object on every render?

sinansonmez replied:
Done

michelengelen reviewed (changes requested):
hi @sinansonmez ... some comments, but overall looks good

michelengelen commented:
@sinansonmez if it is not too much of me to ask could you try and convert the `popover_list_members.tsx` to a functional component as well? It is rather plain and easy to convert.

sinansonmez commented:
> @sinansonmez if it is not too much of me to ask could you try and convert the `popover_list_members.tsx` to a functional component as well? It is rather plain and easy to convert.

Sure @michelengelen . After addressing your points above, I will convert it to functional component

sinansonmez commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -43,70 +42,53 @@ export type Props = {
-            statuses: props.statuses,
-        };
-    }
+    useEffect(() => {
+        setUsers(props.users);
+        setStatuses(props.statuses);
+    }, [props.users, props.statuses]);
```
I suppose this is the correct way to convert `getDerivedStateFromProps`. 

However, I have a doubt if we still need this because I don't see any where `users` or `statuses` states are being used. Please let me know if I am missing sth.

This is the reason of lint failing:
- `  50:12  error  'users' is assigned a value but never used     @typescript-eslint/no-unused-vars`
- `  51:12  error  'statuses' is assigned a value but never used  @typescript-eslint/no-unused-vars`

hmhealey replied:
I think you could actually get rid of both of those from state. I don't know why the old version used `getDerivedStateFromProps` since it doesn't look like it used these either. If I had to guess, it probably predates Redux and just never got cleaned up.

michelengelen replied:
that's what I thought as well ... might be some leftover that never got cleaned up

sinansonmez replied:
Thanks to both of you. 
- `users` is deleted both from `props` and `state`. I also updated index file to remove `users` as we dont need anymore
- `statuses` is deleted from `state` but kept in props

sinansonmez commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -115,201 +97,197 @@ export default class PopoverListMembers extends React.PureComponent<Props, State
+                    id='member_popover'
+                    aria-label={ariaLabel}
+                    className={'member-popover__trigger channel-header__icon channel-header__icon--left channel-header__icon--wide ' + (showPopover ? 'channel-header__icon--active' : '')}
+                    ref={targetRef}
```
Is this the correct way of updating `Ref`?

sinansonmez commented:
@michelengelen Converting to Functional Component was a fun, enlightening and challenging (especially on tests) exercise over the weekend. I have couple clarification points. Please let me know what do you think?

michelengelen reviewed (changes requested):
just one more suggestion to improve types

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,293 @@
+        }
+    };
+
+    const closePopover = () => {
+        setShowPopover(false);
+    };
```
```suggestion
    const closePopover = () => setShowPopover(false);
```

sinansonmez replied:
Done

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,293 @@
+            </div>
+        );
+    }
+    const count = props.memberCount;
+    let countText = '-';
+    if (count > 0) {
+        countText = count.toString();
+    }
```
this is too complicated:

```suggestion
    const countText = props.memberCount > 0 ? props.memberCount.toString() : '-';
```

sinansonmez replied:
Done

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,293 @@
+    ));
+
+    const channelIsArchived = props.channel.delete_at !== 0;
+    let popoverButton;
```
```suggestion
    let popoverButton = null;
```

sinansonmez replied:
Done

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,293 @@
+                    onClick={showMembersModal}
+                >
+                    <i className='icon icon-pencil-outline'/>
+                    <FormattedMessage
+                        id='members_popover.editBtn'
+                        defaultMessage='Edit'
+                    />
+                </button>
+            );
+        } else if (props.addMembersABTest === AddMembersToChanneltreatments.TOP && props.manageMembers) {
+            editButton = (
+                <button
+                    className='btn btn-link'
+                    id='addBtn'
+                    onClick={() => onAddNewMembersButton(AddMembersToChanneltreatments.TOP)}
+                >
+                    <i className='icon icon-account-plus-outline'/>
+                    <FormattedMessage
+                        id='members_popover.add'
+                        defaultMessage='Add'
+                    />
+                </button>
+            );
+        }
+
+        popoverButton = (
+            <div
+                className='more-modal__button'
+                key={'popover-member-more'}
+            >
+                <button
+                    className='btn btn-link'
+                    data-testid='membersModal'
+                    onClick={handleButtonOnClick}
+                >
+                    {membersName}
+                </button>
+            </div>
+        );
+    }
```
this whole section seems a bit too complicated ... could you try and find a way to simplify it? If you need help I can have a look as well, just hit me a DM! :D

sinansonmez replied:
I will look into this and come back to you with a suggestion

michelengelen replied:
if you need help feel free to send me a DM

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,293 @@
+    let handleButtonOnClick = showMembersModal;
+    let editButton = null;
+    if (props.channel.type !== Constants.GM_CHANNEL && !channelIsArchived) {
+        let membersName = (
+            <FormattedMessage
+                id='members_popover.manageMembers'
+                defaultMessage='Manage Members'
+            />
+        );
+
+        const isDefaultChannel = props.channel.name === Constants.DEFAULT_CHANNEL;
+
+        if (isDefaultChannel || !props.manageMembers) {
+            membersName = (
+                <FormattedMessage
+                    id='members_popover.viewMembers'
+                    defaultMessage='View Members'
+                />
+            );
+        }
```
```suggestion
        const isDefaultChannel = props.channel.name === Constants.DEFAULT_CHANNEL;

        const membersName = (isDefaultChannel || !props.manageMembers) ? (
            <FormattedMessage
                id='members_popover.viewMembers'
                defaultMessage='View Members'
            />
        ) : (
            <FormattedMessage
                id='members_popover.manageMembers'
                defaultMessage='Manage Members'
            />
        );
```

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,293 @@
+        setShowPopover(false);
+    };
+
+    const showMembersModal = (e: React.MouseEvent<HTMLButtonElement>) => {
+        e.preventDefault();
+
+        closePopover();
+
+        const modalData = {
+            modalId: ModalIdentifiers.CHANNEL_MEMBERS,
+            dialogProps: props,
+            dialogType: ChannelMembersModal,
+        };
+
+        props.actions.openModal(modalData);
+    };
+
+    const onAddNewMembersButton = (placement: string) => {
+        const {channel, actions} = props;
+        trackEvent('add_members_from_channel_popover', placement);
+
+        actions.openModal({
+            modalId: ModalIdentifiers.CHANNEL_INVITE,
+            dialogType: ChannelInviteModal,
+            dialogProps: {channel},
+        });
+
+        closePopover();
+    };
```
these 2 functions are written differently, although they are basically doing the same thing (with differing data). The question is: could we simplify the functions into one with flagging parameters, or at least built them in a way that they look and work the same?

michelengelen replied:
yeah, but after thinking bout it for a while I would say it would be better to just have them look the same to make them more readable. Making them work with flags might be too much and make them error-prone as well.

sinansonmez replied:
I updated the part related to creating `modalData` and calling `actions.openModal(modalData)` to make them look similar. Please let me know if there is more can be done here

michelengelen reviewed (changes requested):
good work, just a few suggestions

sinansonmez commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -32,15 +32,15 @@ export type Props = {
     actions: {
         openModal: <P>(modalData: ModalData<P>) => void;
-        loadProfilesAndStatusesInChannel: (channelId: string, page: number, perPage: number | undefined, sort: string, options: {active: boolean}) => void;
+        loadProfilesAndStatusesInChannel: (channelId: string, page?: number, perPage?: number, sort?: string, options?: {active: boolean}) => void;
```
I went for the second option :)

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,273 @@
+
+    const closePopover = () => setShowPopover(false);
+
+    const showMembersModal = (e: React.MouseEvent<HTMLButtonElement>) => {
+        const {actions} = props;
+        e.preventDefault();
+        closePopover();
+
+        const modalData = {
+            modalId: ModalIdentifiers.CHANNEL_MEMBERS,
+            dialogProps: props,
+            dialogType: ChannelMembersModal,
+        };
+        actions.openModal(modalData);
+    };
+
+    const onAddNewMembersButton = (placement: string) => {
+        const {channel, actions} = props;
+        trackEvent('add_members_from_channel_popover', placement);
+
+        const modalData = {
+            modalId: ModalIdentifiers.CHANNEL_INVITE,
+            dialogType: ChannelInviteModal,
+            dialogProps: {channel},
+        };
+        actions.openModal(modalData);
+        closePopover();
+    };
```
I would suggest a slight re-structure here as well to make them even more alike

```suggestion
    const showMembersModal = (e: React.MouseEvent<HTMLButtonElement>) => {
        const {actions} = props;

        e.preventDefault();
        closePopover();

        const modalData = {
            modalId: ModalIdentifiers.CHANNEL_MEMBERS,
            dialogProps: props,
            dialogType: ChannelMembersModal,
        };

        actions.openModal(modalData);
    };

    const onAddNewMembersButton = (placement: string) => {
        const {channel, actions} = props;

        trackEvent('add_members_from_channel_popover', placement);

        closePopover();

        const modalData = {
            modalId: ModalIdentifiers.CHANNEL_INVITE,
            dialogType: ChannelInviteModal,
            dialogProps: {channel},
        };

        actions.openModal(modalData);
    };
```

sinansonmez replied:
Done

michelengelen commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,273 @@
+        actions.openModal(modalData);
+    };
+
+    const onAddNewMembersButton = (placement: string) => {
+        const {channel, actions} = props;
+        trackEvent('add_members_from_channel_popover', placement);
+
+        const modalData = {
+            modalId: ModalIdentifiers.CHANNEL_INVITE,
+            dialogType: ChannelInviteModal,
+            dialogProps: {channel},
+        };
+        actions.openModal(modalData);
+        closePopover();
+    };
```
in this case I would suggest a factory style to make constructing it easier in the components. But this step is totally optional:

```suggestion
    const getOnAddNewMembersButton = (placement: string) => (e: React.MouseEvent<HTMLButtonElement>) => {
        const {channel, actions} = props;

        trackEvent('add_members_from_channel_popover', placement);

        e.preventDefault();
        closePopover();

        const modalData = {
            modalId: ModalIdentifiers.CHANNEL_INVITE,
            dialogType: ChannelInviteModal,
            dialogProps: {channel},
        };

        actions.openModal(modalData);
    };
```

that way you can use it in the component like this:

```
...
onClick={getOnAddNewMembersButton(AddMembersToChanneltreatments.BOTTOM)}
...

sinansonmez replied:
Makes sense, done

michelengelen reviewed (commented):
some optional improvements ... nothing serious!

sinansonmez commented on components/popover_list_members/popover_list_members.tsx:
```diff
@@ -0,0 +1,274 @@
+                        <i className='icon icon-pencil-outline'/>
+                        <FormattedMessage
+                            id='members_popover.editBtn'
+                            defaultMessage='Edit'
+                        />
+                    </button>
+                );
+            } else if (props.addMembersABTest === AddMembersToChanneltreatments.TOP) {
+                editButton = (
+                    <button
+                        className='btn btn-link'
+                        id='addBtn'
+                        onClick={getOnAddNewMembersButton(AddMembersToChanneltreatments.TOP)}
+                    >
+                        <i className='icon icon-account-plus-outline'/>
+                        <FormattedMessage
+                            id='members_popover.add'
+                            defaultMessage='Add'
+                        />
+                    </button>
+                );
+            }
+        }
+
+        const handleButtonOnClick = (props.manageMembers && props.addMembersABTest === AddMembersToChanneltreatments.BOTTOM) ? getOnAddNewMembersButton(AddMembersToChanneltreatments.BOTTOM) : showMembersModal;
+        popoverButton = (
+            <div
+                className='more-modal__button'
+                key={'popover-member-more'}
+            >
+                <button
+                    className='btn btn-link'
+                    data-testid='membersModal'
+                    onClick={handleButtonOnClick}
+                >
+                    {membersName}
+                </button>
+            </div>
+        );
+    }
```
Thanks for the suggestion @michelengelen, with two separate functions it is well organized. I committed the changes. Please see my minor comments below. 
- For `renderHeaderButton`, I also thought about using ternary operator. Then I decided to keep as it is (i.e. check explicitly with if `props.addMembersABTest` is `top` or `bottom`. Because (maybe) in some cases `props.addMembersABTest` is something other than `top` or `bottom`. Anyhow, I assume it is safe to just check if it is `top` with ternary operator  
-  For `renderFooterButton`, it is a good idea just to use switch statement. There was a small issue related to one case where  
     - `isDefaultChannel` is `true`
     - `props.manageMembers` is `true`
     -  `props.addMembersABTest === AddMembersToChanneltreatments.BOTTOM`

   In this case `label` should be `Add Members` not `View Members`. I fixed it by changing the order and including `props.manageMembers` in the first case.

sinansonmez commented on components/popover_list_members/index.ts:
```diff
@@ -2,27 +2,43 @@
     return createSelector(
         'makeSortUsersByStatusAndDisplayName',
-        (state, channelId) => doGetProfilesInChannel(state, channelId, true),
+        (state: GlobalState, channelId: Channel['id']) => doGetProfilesInChannel(state, channelId, {active: true}),
```
@hmhealey coming back to your previous comment related to changing from `true` to `{active: true}`. You are right, this will actually change the behavior.  

I was able to track down the usage of `filters` in the following places with this order 
- [makeGetProfilesInChannel](https://github.com/mattermost/mattermost-webapp/blob/7759e71afa0ffa94c4e2bec2e1d09537ae0982e5/packages/mattermost-redux/src/selectors/entities/users.ts#L559) and 
- [filterProfiles](https://github.com/mattermost/mattermost-webapp/blob/7759e71afa0ffa94c4e2bec2e1d09537ae0982e5/packages/mattermost-redux/src/selectors/entities/users.ts#L281)

As `filters` is expecting an object, so we can't just pass `true` but we can't also pass `{active: true}` unless we want to change the behavior. I assume when we pass just `true` this has no effect. 

Therefore we can just remove `{active: true}` not pass anything as default value for `filters` is `{}` on [makeGetProfilesInChannel](https://github.com/mattermost/mattermost-webapp/blob/7759e71afa0ffa94c4e2bec2e1d09537ae0982e5/packages/mattermost-redux/src/selectors/entities/users.ts#L559) and it is optional in [filterProfiles](https://github.com/mattermost/mattermost-webapp/blob/7759e71afa0ffa94c4e2bec2e1d09537ae0982e5/packages/mattermost-redux/src/selectors/entities/users.ts#L281)

Please let me know what you think on this?

hmhealey replied:
Yeah, I think passing nothing would work best. That'll also work around an issue I noticed in `doGetProfilesInChannel` where it doesn't handle a filters being an object properly, causing it to do a bunch of unnecessary background work. That's a separate bug to fix though

sinansonmez replied:
Agreed, I did the last commit

michelengelen reviewed (approved):
LGTM ... nice work! ✨

jgilliam17 commented:
/update-branch

jgilliam17 commented:
E2Es running [here](https://automation-dashboard.vercel.app/cycles/32a597c9-2c26-4054-ad0d-4eda7b20b47e)

jgilliam17 commented:
Thanks @sinansonmez 
Tested, looks good to merge.
- Verified member list modals - working as expected 
- [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/174975-bb2d74e-onprem-ent-webapp-pr-10025-mm-ee-test:bb2d74e/mochawesome.html) looks good, no PR related failures.
